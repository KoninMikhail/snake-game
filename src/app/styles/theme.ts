import {colors, ThemeColors} from './theme_parts/colors';
import {gaps, ThemeGaps} from './theme_parts/gaps';
import {icons, ThemeIcons} from './theme_parts/icons';
import {ThemeTypography, typography} from './theme_parts/typography';
import {animations, ThemeAnimations} from '@styles/theme_parts/animations';

/**
 * @typedef IBaseTheme
 */
export interface IBaseTheme {
    animations: ThemeAnimations;
    colors: ThemeColors;
    gaps: ThemeGaps;
    icons: ThemeIcons;
    typography: ThemeTypography;
}

/**
 * Theme
 */
export const theme: IBaseTheme = {
    colors,
    animations,
    gaps,
    icons,
    typography,
};
