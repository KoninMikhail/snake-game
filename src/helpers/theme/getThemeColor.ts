import { ThemeColors } from '@styles/theme_parts/colors';
import { IBaseTheme } from '@styles/theme';

/**
 * @name getThemeColor
 *
 * Get the color from theme.
 *
 * @param theme - The theme object
 * @param color - The color you want to get from the theme.
 * @returns A function that takes a theme and a color and returns the color from the theme.
 */
const getThemeColor: GetFromThemeFn<IBaseTheme, ThemeColors> = (theme, color) => {
    return theme.colors[color];
};

export default getThemeColor;
