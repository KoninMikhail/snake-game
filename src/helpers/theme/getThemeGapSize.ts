import {IBaseTheme} from '@styles/theme';
import {ThemeGaps} from '@styles/theme_parts/gaps';

/**
 * @name getThemeGapSize
 *
 * Get the gap size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the gap you want to get.
 * @returns A function that takes a theme and a size and returns the theme.gaps[size]
 */
const getThemeGapSize: GetFromThemeFn<IBaseTheme, ThemeGaps> = (theme, size) => theme.gaps[size];

export default getThemeGapSize;
