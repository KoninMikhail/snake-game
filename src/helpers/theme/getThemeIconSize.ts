import { IBaseTheme } from '@styles/theme';
import { ThemeIcons } from '@styles/theme_parts/icons';

/**
 * @name getThemeIconSize
 *
 * Get the icon size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the icon you want to get.
 * @returns A function that takes a theme and a size and returns the theme.icons[size]
 */
const getThemeIconSize: GetFromThemeFn<IBaseTheme, ThemeIcons> = (theme, size) => {
    return theme.icons[size];
};

export default getThemeIconSize;
