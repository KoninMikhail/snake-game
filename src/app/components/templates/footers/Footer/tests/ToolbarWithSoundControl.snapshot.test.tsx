import {render} from '@testing-library/react';
import {ThemeProvider} from 'styled-components';
import {IBaseTheme} from '@styles/theme';
import {
    ToolbarWithSoundControlTemplate
} from '@templates/toolbars/ToolbarWithSoundControl/ToolbarWithSoundControlTemplate';
import {mockViewport} from 'jsdom-testing-mocks';

/**
 * Mocks
 */
// Fake theme
const fakeTheme = {
    icons: {
        xl: '3.6em',
    },
    gaps: {
        md: '1rem',
    },
} as unknown as IBaseTheme;

// fake content
const mockSoundContent = 'mocked SoundControl';
const mockLevelContent = 'mocked Level';

// fake component
jest.mock('@ui/organisms/SoundControl/SoundControl', () => ({
    SoundControl: () => mockSoundContent,
}));
jest.mock('@ui/molecules/information/Level/Level', () => ({
    Level: () => mockLevelContent,
}));

/**
 * Tests
 */
describe('ToolbarWithSoundControlDesktop', () => {
    it('Equals snapshot on mobile devices', () => {
        mockViewport({ width: '320px', height: '568px' });

        const { container } = render(
            <ThemeProvider theme={fakeTheme}>
                <ToolbarWithSoundControlTemplate />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });

    it('Equals snapshot on tablet devices', () => {
        mockViewport({ width: '885px', height: '768px' });

        const { container } = render(
            <ThemeProvider theme={fakeTheme}>
                <ToolbarWithSoundControlTemplate />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
    it('Equals snapshot on desktop devices', () => {
        mockViewport({ width: '1920px', height: '1080px' });

        const { container } = render(
            <ThemeProvider theme={fakeTheme}>
                <ToolbarWithSoundControlTemplate />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
