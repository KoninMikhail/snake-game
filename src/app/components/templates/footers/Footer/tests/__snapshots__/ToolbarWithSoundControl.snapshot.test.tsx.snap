// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarWithSoundControlDesktop Equals snapshot on desktop devices 1`] = `
.c0 {
  display: block;
  width: 100%;
}

.c2 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: end;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
  gap: 1rem;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        mocked SoundControl
      </div>
    </div>
  </header>
</div>
`;

exports[`ToolbarWithSoundControlDesktop Equals snapshot on mobile devices 1`] = `
.c0 {
  display: block;
  width: 100%;
}

.c2 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: end;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
  gap: 1rem;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        mocked SoundControl
      </div>
    </div>
  </header>
</div>
`;

exports[`ToolbarWithSoundControlDesktop Equals snapshot on tablet devices 1`] = `
.c0 {
  display: block;
  width: 100%;
}

.c2 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: end;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
  gap: 1rem;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        mocked SoundControl
      </div>
    </div>
  </header>
</div>
`;
